tests/pipeline/test_blaze.py

```diff
@@ -1128,3 +1131,82 @@ def test_novel_deltas_macro(self):
+        missing_sid = ord('E')
+        handler = make_test_handler(self)
+        asset_info = asset_infos[0][0]
+        base_dates = pd.DatetimeIndex([
+            pd.Timestamp('2014-01-01'),
+            pd.Timestamp('2014-01-04')
+        ])
+        repeated_dates = base_dates.repeat(4)
+        baseline = pd.DataFrame({
+            'sid': (self.sids + (missing_sid,)) * 2,
+            'value': (0., 1., 2., 3., 1., 2., 3., 4.),
+            'int_value': (0, 1, 2, 3, 1, 2, 3, 4.),
+            'asof_date': repeated_dates,
+            'timestamp': repeated_dates
+        })
+        expr = bz.Data(baseline, name='expr', dshape=self.dshape)
+        deltas = bz.Data(
+            odo(
+                bz.transform(
+                    expr,
+                    value=expr.value + 10,
+                    timestamp=expr.timestamp + timedelta(days=1),
+                ),
+                pd.DataFrame,
+            ),
+            name='delta',
+            dshape=self.dshape,
+        )
+        expected_views = keymap(pd.Timestamp, {
+            '2014-01-03': np.array([[10.0, 11.0, 12.0],
+                                    [10.0, 11.0, 12.0],
+                                    [10.0, 11.0, 12.0]]),
+            '2014-01-06': np.array([[10.0, 11.0, 12.0],
+                                    [10.0, 11.0, 12.0],
+                                    [11.0, 12.0, 13.0]]),
+        })
+        if len(asset_info) == 4:
+            expected_views = valmap(
+                lambda view: np.c_[view, [np.nan, np.nan, np.nan]],
+                expected_views,
+            )
+            expected_output_buffer = [10, 11, 12, np.nan, 11, 12, 13, np.nan]
+        else:
+            expected_output_buffer = [10, 11, 12, 11, 12, 13]
+
+        cal = pd.DatetimeIndex([
+            pd.Timestamp('2014-01-01'),
+            pd.Timestamp('2014-01-02'),
+            pd.Timestamp('2014-01-03'),
+            # omitting the 4th and 5th to simulate a weekend
+            pd.Timestamp('2014-01-06'),
+        ])
+        with handler.applicationbound():
+            with tmp_asset_finder(equities=asset_info) as finder:
+                expected_output = pd.DataFrame(
+                    expected_output_buffer,
+                    index=pd.MultiIndex.from_product((
+                        sorted(expected_views.keys()),
+                        finder.retrieve_all(asset_info.index),
+                    )),
+                    columns=('value',),
+                )
+                self._run_pipeline(
+                    expr,
+                    deltas,
+                    expected_views,
+                    expected_output,
+                    finder,
+                    calendar=cal,
+                    start=cal[2],
+                    end=cal[-1],
+                    window_length=3,
+                    compute_fn=op.itemgetter(-1),
+                )
+            message = handler.records[0].message
```

Comment:
we should scan the records to see if exactly one matches the pattern because this might not be the first thing logged.